common/lib/dependabot/shared_helpers.rb

```diff
@@ -262,17 +265,18 @@ def self.reset_global_git_config(backup_path)
       FileUtils.mv(backup_path, GIT_CONFIG_GLOBAL_PATH)
     end
 
-    def self.run_shell_command(command)
+    def self.run_shell_command(command, escape_command_str: true)
```

Comment:
I think this is good as is, but I was wondering if a method name that's very explicit about this being unsafe would be even better?

```
def self.run_unsafe_shell_command(command)
```

Reply:
👍🏻  to the above, I just wanted to follow up - is there ever any reason we should distinguish between safe/unsafe even in hard-coded internal calls?

My instinct would be to default to principal of least trust at all times _just in case_. We could indirectly allow unsafe strings to get into a 'trusted' `run_shell_command` as a parameter of another method without being able to spot it easily.

Reply:
Pushed a commit to rename the argument `escape_command_str` to `allow_unsafe_shell_command`: https://github.com/dependabot/dependabot-core/pull/2725/commits/b36a5a0d972a51954b1a913c4c46275a5ee67a25

Thought on this?

> We could indirectly allow unsafe strings to get into a 'trusted' run_shell_command as a parameter of another method without being able to spot it easily. 

Not sure I follow, did you have an example in mind given the current design?

Reply:
I think the last commit solves the thing that made me itchy, I was looking at the calls in `configure_git_to_use_https`, e.g.

```
      run_shell_command(
        "git config --global --add url.https://github.com/."\
        "insteadOf git@github.com:"
      )
```

My thought was that if this method was ever parameterised (e.g. force https access to a private source ) the author/reviewer would need to be diligent to consider escaping these strings.

The change to break that method up into several calls to `run_shell_command` and making it escape by default limits the risk to configuring the credentials [here](https://github.com/dependabot/dependabot-core/pull/2725/files#diff-219bebe959e95ebf38d9a9fa9225f9df2cf12195eab2ee0e04054672ff178a0dR210)

That covers what I was concerned about well 👍🏻

Reply:
> My thought was that if this method was ever parameterised

Yep, good call 👍 

> That covers what I was concerned about well 👍🏻

Thanks for checking 👌